File: js/숫자문자열과 영단어.js
function solution(s) {
    const num_dic = {"zero":"0", "one":"1", "two":"2", "three":"3", "four":"4", "five":"5", "six":"6", "seven":"7", "eight":"8", "nine":"9"};
    var answer = s;
    for (value in num_dic) {
        // 주의: replace를 쓰면 중복된 단어가 나올 때 앞의 하나만 바뀌게 됨. 따라서 replaceAll 사용
        answer = answer.replaceAll(value, num_dic[value]);
    }
    return Number(answer);
}

File: python/숫자 문자열과 영단어.py
# def solution(s):
#     arr = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight','nine']
#     tmp = ''
#     tmp2 = ''
#     for elem in s:
#         if elem.isdigit():
#             tmp += elem
#         else:
#             tmp2 += elem
#             if tmp2 in arr:
#                 tmp += str(arr.index(tmp2))
#                 tmp2 = ''

#     return int(tmp)
        
num_dic = {"zero":"0", "one":"1", "two":"2", "three":"3", "four":"4", "five":"5", "six":"6", "seven":"7", "eight":"8", "nine":"9"}

def solution(s):
    answer = s
    for key, value in num_dic.items():
        answer = answer.replace(key, value)
    return int(answer)

